pallets/msa/src/offchain_storage.rs

```diff
@@ -0,0 +1,181 @@
+/// Offchain Storage for MSA
+use codec::{Decode, Encode};
+use common_primitives::offchain as offchain_common;
+use frame_support::log::error as log_err;
+use sp_runtime::offchain::storage::StorageRetrievalError;
+use sp_std::{
+	fmt::{Debug, Formatter},
+	vec::Vec,
+};
+
+/// Block event storage prefix
+pub const BLOCK_EVENT_KEY: &[u8] = b"frequency::block_event::msa::";
+
+/// Generic prefix for MSA index storage
+pub const MSA_INDEX_KEY: &[u8] = b"frequency::msa::";
+
```

Comment:
Prefixes used in this implementation one to index `BLOCK_EVENT_KEY` to offchain_index and `MSA_INDEX_KEY` to store and retrieve public keys for a given msa_id